### PR TITLE
mrc-5439: Update vignette, regen docs, remove usage of leapfrog

### PR DESCRIPTION
**Pull request by r-ash**
This PR does some work to remove leapfrog dependency in prep for replacing leapfrog with frogger
* Add content from leapfrog README into this one. Note this uses a Rmd README, you'll need to run `devtools::build_readme()` if you make any changes to the README. I will add a CI script to do this automatically in the future.
* Moves the data prep commands we need for the README into the package
* Update documentation `devtools::document()`
* Tidy some R CMD warnings and notes (e.g. using beers package over eppasm:::create_beers)
* Run code generation as part of package build and installation via `configure` script

Left to do here before this is ready to replace leapfrog is
1. Review this PR, is there anything from leapfrog I have missed?
2. Move leapfrog to a new location, leapfrog.old? leapfrog.original? Open for suggestions. leapfrog.archived
3. Update the `create_test_data.R` script and reference tests. We still have tests which compare this output to leapfrog output.
   1. Are we happy with the tests here to remove the leapfrog reference tests?
   2. We could keep them but remove them from `create_test_data.R` so they won't be updated anymore, at least until the impl here for the adult model diverges
   3. We could update `create_test_data.R` to use the moved leapfrog, `leapfrog.original`. Again can keep this until the model diverges from what exists there
4. Bulk renaming, rename files here to leapfrog from frogger, rename package, remove all references to `frogger` as a name.
(5a. Squash the history -- this would remove all the data churn which is currently bloating this package a bit, but we would lose the history, undecided about this)
5. Move frogger to leapfrog

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,23 +1,40 @@
...
+           role = c("aut"),
+           email = "m.walters@imperial.ac.uk"),
+    person(given = "Robert", 
+           family = "Ashton", 
+           role = c("aut", "cre"),
+           email = "robertashton94@gmail.com"),
+    person(given = "Mantra", 
+           family = "Kusumgar", 
+           role = c("aut"),
+           email = "m.kusumgar@imperial.ac.uk"))
```
Comment (r-ash) [inline DESCRIPTION:21]: Let me know if you want a different email here or anything. I've put myself as the maintainer but we can change that if we want.. not really sure who to put down. Sadly we can only have 1

File: LICENSE [hunk 2]
```diff
@@ -1,2 +1,2 @@
-YEAR: 2021
-COPYRIGHT HOLDER: Imperial College of Science, Technology and Medicine
+YEAR: 2024
+COPYRIGHT HOLDER: Jeff Eaton
```
Comment (r-ash) [inline LICENSE:2]: Porting this over from leapfrog

File: R/generate_cpp.R [hunk 3]
```diff
@@ -362,18 +362,6 @@ generate_struct_def <- function(inputs) {
...
-  paste0(
-    sprintf(
-      "  const leapfrog::%s<real_type> %s_params = {\n",
-      inputs[[1]]$struct,
-      to_lower_camel(inputs[[1]]$struct)
-    ),
-    paste_lines(sprintf("    %s", vcapply(inputs, "[[", "cpp_name"))),
-    "  \n};\n"
-  )
-}
```
Comment (r-ash) [inline R/generate_cpp.R:375]: Turns out this wasn't being used

File: README.Rmd [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+
+<!-- badges: start -->
+[![Project Status: Concept – Minimal or no implementation has been done yet, or the repository is only intended to be a limited example, demo, or proof-of-concept.](https://www.repostatus.org/badges/latest/concept.svg)](https://www.repostatus.org/#concept)
+[![R build status](https://github.com/mrc-ide/frogger/workflows/R-CMD-check/badge.svg)](https://github.com/mrc-ide/frogger/actions)
+[![codecov.io](https://codecov.io/github/mrc-ide/frogger/coverage.svg?branch=main)](https://codecov.io/github/mrc-ide/frogger?branch=main)
+<!-- badges: end -->
+
+Leapfrog is a multistate population projection model for demographic and HIV epidemic estimation.
+
+The name _leapfrog_ is in honor of [Professor](https://iussp.org/en/basia-zaba-1949-2018) Basia [Zaba](https://translate.google.co.uk/?sl=pl&tl=en&text=Zaba&op=translate).
```
Comment (r-ash) [inline README.Rmd:26]: Note the updates URLs here, looks like original ones from leapfrog no longer work :(

File: DESCRIPTION (hunk 1 above)
Comment (mwalte10) [reply DESCRIPTION:21]: Assuming Jeff wants Imai-Eaton as well

Comment (r-ash) [reply DESCRIPTION]: Thanks, I've updated those

File: R/spectrum_inputs.R [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+  idx_expand_coarse <- rep(1:4, times = c(3, 2, 2, 2))
+
+  v <- list()
+  v$incidinput <- eppasm::read_incid_input(pjnz)
+  v$incidpopage <- attr(v$incidinput, "incidpopage")
+  v$incrr_sex <- projp$incrr_sex
+
+  ## HIV effects on fertilty
+  # x <- c(6.663660,	6.627490,	6.587540,	6.550000,	6.515790,	6.480510,	6.442340,	6.399450,	6.350000,	6.292810,	6.226410,	6.148620,	6.057220,	5.950000,	5.795050,	5.582180,	5.341780,	5.104260,	4.900000,	4.728830,	4.568750,	4.417400,	4.272430,	4.131500,	3.988570,	3.844730,	3.708350,	3.587750,	3.491300,	3.414530,	3.346940,	3.287800,	3.236360,	3.191900,	3.150350,	3.110410,	3.075540,	3.049190,	3.034800,	3.029150,	3.025720,	3.023080,	3.019750,	3.014300,	3.002190,	2.981570,	2.955020,	2.925100,	2.894400,	2.859560,	2.817970,	2.773350,	2.729390,	2.689800,	2.654560,	2.621020,	2.588890,	2.557880,	2.527700,	2.498230,	2.469570)
+  # v$tfr = x
```
Comment (mwalte10) [inline R/spectrum_inputs.R:185]: this can be deleted

File: R/spectrum_inputs.R [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+  v$adult_cd4_dist <- adult_cd4_dist_array
+
+
+  v$mtct_trans <- (c(0.15, 0.15, 0.27, 0.27, 0.37, 0.37, 0.37))
+
+  v$fert_mult_by_age <- rep(c(1.153260, 1.001870, 0.909590,0.912760, 0.883990, 0.883990, 0.883990), each = 5)
+
+  v$fert_mult_offart <- c(1, 0.96, 0.88, 0.78, 0.61, 0.38, 0.30)
+
+  v$fert_mult_onart <- rep(c(1.094460, 1.006870, 0.916170, 0.810910, 0.637220, 0.637220, 0.637220), each = 5)
```
Comment (mwalte10) [inline R/spectrum_inputs.R:207]: I need to change this so that it is pulling from the DP file not hardcoded here

File: scripts/read_spectrum.R [hunk 7]
```diff
@@ -1,3 +1,303 @@
...
+  ## 12+ mo on treatment
+  adol_art_mort[,3,1:5] <- mort$hc2_gte12[,1,1]
+  adol_art_mort[,3,6:10] <- mort$hc2_gte12[,2,1]
+
+  v$paed_cd4_mort <- paed_cd4_mort
+  v$adol_cd4_mort <- adol_cd4_mort
+  v$paed_art_mort <- paed_art_mort
+  v$adol_art_mort <- adol_art_mort
+
+  mtct_rates_input <- dp_read_mtct_rates(pjnz)
```
Comment (mwalte10) [inline scripts/read_spectrum.R:87]: These might be in leapfrog...

Comment (r-ash) [reply scripts/read_spectrum.R:87]: I've ported these over, on line 723

**Pull request merged**